UIComponents/UIComponents/Helper/CLLocationHelper.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  CLLocationHelper.swift
+//  UIComponents
+//
+//  Created by Sagar Ajudiya on 26/05/23.
+//
+
+import CoreLocation
+
+class CLLocationHelper: NSObject {
+
+    private override init() {
+        super.init()
+        checkLocationAuth()
+        locationManager.delegate = self
+        locationManager.desiredAccuracy = kCLLocationAccuracyBest
+        locationManager.distanceFilter = 100.0
+    }
+
+    // MARK: Variables
+    static let shared = CLLocationHelper()
+    private let locationManager = CLLocationManager()
+    
+    var locationCompletionHandler: ((CLLocation?) -> Void)?
+
+    // iPhone location service check
+    func checkLocationServices() {
+        DispatchQueue.global().async {
```

Comment:
Here i taken global because in global task perform in any available thread and if i take main then app will take more time if main thread is not available at that time